scripts/setup-minikube.sh

```diff
@@ -0,0 +1,354 @@
+  echo -e "Minikube dashboard at ${GREEN}http://$minikube_ip:30000/${NC}"
+}
+
+function infer_minikube_settings() {
+  # cpus
+  local np=3
+  if [[ $PLATFORM == $LINUX ]]
+  then
+    np=`nproc --all`
+  elif [[ $PLATFORM == $OSX ]]
+  then
+    np=`sysctl -n hw.ncpu`
+  fi
+  echo -n "Assign CPUs between 3..$(( np - 2 )) [$(( np / 2 ))]: "
+  read p
+  if [[ $p == "" ]]
+  then
+    p=$(( np / 2 ))
+  fi
+  MINIKUBE_CMD=$MINIKUBE_CMD" --cpus $p"
+
+  # memory
+  local mm=8
+  if [[ $PLATFORM == $LINUX ]]
+  then
+    local mm_kb=`cat /proc/meminfo | grep 'MemTotal' | awk '{print $2}'`
+    mm=$(( mm_kb * KB1 / GB1 ))
+  elif [[ $PLATFORM == $OSX ]]
+  then
+    mm=$(( `sysctl -n hw.memsize` / GB1 ))
+  fi
+  if (( $mm < 16 ))
+  then
+    echo -e "${YELLOW}WARNING: Your system has low memory for an ML workload.${NC}"
+  fi
+  echo -n "Assign memory (in GB) between 8..$(( mm - 5 )) [$(( mm / 2 ))]: "
+  read m
+  if [[ $m == "" ]]
+  then
+    m=$(( mm / 2 ))
+  fi
+  MINIKUBE_CMD=$MINIKUBE_CMD" --memory $(( m * KB1 ))"
+
+  # disk
+  local dd=40
+  dd=`df -h . | awk '{print $4}' | tail -1 | sed 's/[A-Za-z]//g'`
+  if (( $dd < 40 ))
+  then
+    echo -e "${YELLOW}WARNING: Low available disk space.${NC}"
+  fi
+  echo -n "Assign disk space (in GB) between 40..$(( dd / 2 )) [50]: "
+  read d
+  if [[ $d == "" ]]
+  then
+    d=50
+  fi
+  MINIKUBE_CMD=$MINIKUBE_CMD" --disk-size ${d}g"
+
+  # virtualizer
+  echo -n "Choose a virtualizer/hypervisor installed on your system (virtualbox, vmwarefusion, kvm2, hyperkit): "
+  read v
+  if [[ $v != "" ]]
+  then
+    MINIKUBE_CMD=$MINIKUBE_CMD" --vm-driver=$v"
+  fi
+
+  # Mount local dir
+  echo -n "If you'd like to access a local directory in JupyterHub, please enter the full path: "
+  read l
+  if [[ $l != "" ]]
+  then
+    MINIKUBE_CMD=$MINIKUBE_CMD" --mount-string=$l:/mnt/local --mount"
+    MOUNT_LOCAL=true
+  fi
+}
+
+function download_kfctl_scripts() {
+  curl -O https://raw.githubusercontent.com/kubeflow/kubeflow/master/scripts/kfctl.sh
+  chmod +x kfctl.sh
+  curl -O https://raw.githubusercontent.com/kubeflow/kubeflow/master/scripts/util.sh
+  mkdir -p gke
+  pushd .
+  cd gke
+  curl -O https://raw.githubusercontent.com/kubeflow/kubeflow/master/scripts/gke/util.sh
+  popd
+}
+
+function deploy_kubeflow() {
+  # pull bootstrapper config and deloy kubeflow
+  # curl -O https://raw.githubusercontent.com/kubeflow/kubeflow/v0.2.0/bootstrap/bootstrapper.yaml
+  # kubectl create -f bootstrapper.yaml
+
+  # kfctl installation
+  set +e
+  O=`kubectl get namespace kubeflow 2>&1`
+  RESULT=$?
+  set -e
+
+  if [ "${RESULT}" -eq 0 ]; then
+    echo namespace kubeflow already exists
+  else
+    kubectl create namespace kubeflow
+  fi
+
+  #download_kfctl_scripts
+  KUBEFLOW_REPO=/usr/local/google/home/abhishek/kubeflow ./kfctl.sh init localapp --platform minikube
+  pushd .
+  cd localapp
+  ../kfctl.sh generate all
+  ../kfctl.sh apply all
+  popd
+
+  if is_kubeflow_ready
```

Comment:
Per other comments; if we call this from kfctl.sh then most of the functionality up to this line should already be handled by kfctl.sh